### ISSUE TITLE
Finetuning Kokoro

### ISSUE THREAD

**Issue by HURIMOZ** [hexgrad/kokoro#207]
Hi, can we finetune Kokoro with our own text-audio datasets for languages that are not yet existent in Kokoro?
If so, what is the format of the prompts? Is there any config template we could start with?

**Comment by almugabo** [comment]
this indeed would be a great thing to have.

**Comment by DiamondGlassDrill** [comment]
Would love as well being able to support adding more languages to Kokoro let me know how to do it @hexgrad

**Comment by BasToTheMax** [comment]
+1

**Comment by Nikola-Milovic** [comment]
No need to comment +1, usually the etiquette is to just thumbs up the issue. This prevents unnecessary pinging and notifications. You can follow the issue on the side with a button and get notified without commenting

**Comment by Amirjab21** [comment]
It seems to me that the code for the model is available on the repo - so is the only thing remaining for someone to actually build a fine-tuning script? If so, I can do this.

Edit: I've built a small training loop with a very simple loss function that compares a mel spectogram of the reference audio and the generated one. Unfortunately, the loss function you'd need for this model is way more complicated. If it's been trained in a similar way to TTS2, then it requires multi-step training with a GAN.

I will keep looking at it but I've never trained a model this complex before so I'm not sure how far I'll get (plus from my understanding, this would require fairly significant GPU resources)

**Comment by jonirajala** [comment]
> It seems to me that the code for the model is available on the repo - so is the only thing remaining for someone to actually build a fine-tuning script? If so, I can do this.
> 
> Edit: I've built a small training loop with a very simple loss function that compares a mel spectogram of the reference audio and the generated one. Unfortunately, the loss function you'd need for this model is way more complicated. If it's been trained in a similar way to TTS2, then it requires multi-step training with a GAN.
> 
> I will keep looking at it but I've never trained a model this complex before so I'm not sure how far I'll get (plus from my understanding, this would require fairly significant GPU resources)

Im also very interested in this. Have you made any progress in this?

**Comment by info-wordcab** [comment]
Found this implementation, didn't get it to work: https://github.com/igorshmukler/kokoro-ruslan

**Comment by jonirajala** [comment]
Based on that i started building an english version with ljspeech https://github.com/jonirajala/kokoko_training

The architecture from kokoro_ruslan wasnt quite like kokoro. Got that learning and will next try to modify it to be like actual kokoro.

**Comment by codename0og** [comment]
@Amirjab21 @jonirajala 
Hey folks!
Any updates on whether you were able to get finetuning going or not?
Thanks in advance!

**Comment by thewh1teagle** [comment]
https://github.com/semidark/kikiri-tts

**Comment by HURIMOZ** [comment]
> https://github.com/semidark/kikiri-tts

Will give it a try soon. Thanks for sharing.

**Comment by xhy2008** [comment]
> https://github.com/semidark/kikiri-tts

This is still too consuming.I just want a new voice instead of a new language.
Is it possible to finetune the voices.pt only, not the whole model?
